TestWSClientsLimit fails macos

[Run tests (macos-14, 1.22)](https://github.com/nspcc-dev/neo-go/actions/runs/8472779511/job/23217827420#logs)
[6_Run tests (windows-2022, 1.22).txt](https://github.com/nspcc-dev/neo-go/files/14800506/6_Run.tests.windows-2022.1.22.txt)

```
2024-03-28T20:02:13.6376650Z === RUN   TestWSClientsLimit
2024-03-28T20:02:13.6376780Z === RUN   TestWSClientsLimit/default
2024-03-28T20:02:13.6377380Z     logger.go:146: 2024-03-28T19:59:33.367Z	INFO	initial gas supply is not set or wrong, setting default value	{"InitialGASSupply": "52000000"}
2024-03-28T20:02:13.6378100Z     logger.go:146: 2024-03-28T19:59:33.367Z	INFO	P2PNotaryRequestPayloadPool size is not set or wrong, setting default value	{"P2PNotaryRequestPayloadPoolSize": 1000}
2024-03-28T20:02:13.6378610Z     logger.go:146: 2024-03-28T19:59:33.367Z	INFO	MaxBlockSize is not set or wrong, setting default value	{"MaxBlockSize": 262144}
2024-03-28T20:02:13.6379180Z     logger.go:146: 2024-03-28T19:59:33.368Z	INFO	MaxBlockSystemFee is not set or wrong, setting default value	{"MaxBlockSystemFee": 900000000000}
2024-03-28T20:02:13.6379790Z     logger.go:146: 2024-03-28T19:59:33.368Z	INFO	MaxTransactionsPerBlock is not set or wrong, using default value	{"MaxTransactionsPerBlock": 512}
2024-03-28T20:02:13.6380420Z     logger.go:146: 2024-03-28T19:59:33.368Z	INFO	MaxValidUntilBlockIncrement is not set or wrong, using default value	{"MaxValidUntilBlockIncrement": 5760}
2024-03-28T20:02:13.6380830Z     logger.go:146: 2024-03-28T19:59:33.369Z	INFO	no storage version found! creating genesis block
2024-03-28T20:02:13.6381380Z     logger.go:146: 2024-03-28T19:59:33.372Z	INFO	ExtensiblePoolSize is not set or wrong, using default value	{"ExtensiblePoolSize": 20}
2024-03-28T20:02:13.6381900Z     logger.go:146: 2024-03-28T19:59:33.372Z	INFO	SessionPoolSize is not set or wrong, setting default value	{"SessionPoolSize": 20}
2024-03-28T20:02:13.6382490Z     logger.go:146: 2024-03-28T19:59:33.372Z	INFO	MaxIteratorResultItems is not set or wrong, setting default value	{"MaxIteratorResultItems": 100}
2024-03-28T20:02:13.6383040Z     logger.go:146: 2024-03-28T19:59:33.372Z	INFO	MaxFindResultItems is not set or wrong, setting default value	{"MaxFindResultItems": 100}
2024-03-28T20:02:13.6383560Z     logger.go:146: 2024-03-28T19:59:33.372Z	INFO	MaxNEP11Tokens is not set or wrong, setting default value	{"MaxNEP11Tokens": 100}
2024-03-28T20:02:13.6384150Z     logger.go:146: 2024-03-28T19:59:33.372Z	INFO	MaxRequestBodyBytes is not set or wong, setting default value	{"MaxRequestBodyBytes": 5242880}
2024-03-28T20:02:13.6384730Z     logger.go:146: 2024-03-28T19:59:33.372Z	INFO	MaxRequestHeaderBytes is not set or wong, setting default value	{"MaxRequestHeaderBytes": 1048576}
2024-03-28T20:02:13.6385300Z     logger.go:146: 2024-03-28T19:59:33.372Z	INFO	MaxWebSocketClients is not set or wrong, setting default value	{"MaxWebSocketClients": 64}
2024-03-28T20:02:13.6385670Z     logger.go:146: 2024-03-28T19:59:33.372Z	INFO	starting rpc-server	{"endpoint": "127.0.0.1:0"}
2024-03-28T20:02:13.6386120Z     logger.go:146: 2024-03-28T19:59:33.923Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6386710Z     logger.go:146: 2024-03-28T19:59:34.465Z	INFO	persisted to disk	{"blocks": 0, "keys": 127, "headerHeight": 0, "blockHeight": 0, "took": "209.959µs"}
2024-03-28T20:02:13.6387150Z     logger.go:146: 2024-03-28T19:59:35.144Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6387590Z     logger.go:146: 2024-03-28T19:59:35.792Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6388020Z     logger.go:146: 2024-03-28T19:59:36.763Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6388470Z     logger.go:146: 2024-03-28T19:59:37.379Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6388910Z     logger.go:146: 2024-03-28T19:59:38.132Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6389350Z     logger.go:146: 2024-03-28T19:59:39.158Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6389780Z     logger.go:146: 2024-03-28T19:59:40.103Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6390310Z     logger.go:146: 2024-03-28T19:59:41.164Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6390840Z     logger.go:146: 2024-03-28T19:59:42.245Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6391270Z     logger.go:146: 2024-03-28T19:59:43.296Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6391700Z     logger.go:146: 2024-03-28T19:59:43.874Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6392150Z     logger.go:146: 2024-03-28T19:59:44.953Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6392590Z     logger.go:146: 2024-03-28T19:59:45.826Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6393020Z     logger.go:146: 2024-03-28T19:59:46.845Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6393460Z     logger.go:146: 2024-03-28T19:59:48.230Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6393900Z     logger.go:146: 2024-03-28T19:59:49.314Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6394340Z     logger.go:146: 2024-03-28T19:59:50.440Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6394780Z     logger.go:146: 2024-03-28T19:59:51.624Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6395210Z     logger.go:146: 2024-03-28T19:59:52.578Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6395630Z     logger.go:146: 2024-03-28T19:59:53.462Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6396050Z     logger.go:146: 2024-03-28T19:59:54.309Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6396470Z     logger.go:146: 2024-03-28T19:59:55.485Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6396890Z     logger.go:146: 2024-03-28T19:59:56.219Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6397310Z     logger.go:146: 2024-03-28T19:59:57.324Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6397730Z     logger.go:146: 2024-03-28T19:59:58.361Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6398150Z     logger.go:146: 2024-03-28T19:59:59.177Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6398560Z     logger.go:146: 2024-03-28T20:00:00.688Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6398980Z     logger.go:146: 2024-03-28T20:00:01.639Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6399400Z     logger.go:146: 2024-03-28T20:00:02.891Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6399830Z     logger.go:146: 2024-03-28T20:00:03.665Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6400250Z     logger.go:146: 2024-03-28T20:00:04.664Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6400660Z     logger.go:146: 2024-03-28T20:00:05.279Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6401080Z     logger.go:146: 2024-03-28T20:00:06.320Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6401500Z     logger.go:146: 2024-03-28T20:00:07.234Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6401920Z     logger.go:146: 2024-03-28T20:00:08.047Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6402390Z     logger.go:146: 2024-03-28T20:00:09.158Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6402890Z     logger.go:146: 2024-03-28T20:00:10.225Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6403360Z     logger.go:146: 2024-03-28T20:00:11.298Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6403780Z     logger.go:146: 2024-03-28T20:00:12.333Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6404200Z     logger.go:146: 2024-03-28T20:00:13.362Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6404620Z     logger.go:146: 2024-03-28T20:00:14.013Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6405030Z     logger.go:146: 2024-03-28T20:00:15.032Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6405450Z     logger.go:146: 2024-03-28T20:00:16.010Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6405870Z     logger.go:146: 2024-03-28T20:00:17.075Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6406290Z     logger.go:146: 2024-03-28T20:00:17.700Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6406710Z     logger.go:146: 2024-03-28T20:00:18.617Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6407130Z     logger.go:146: 2024-03-28T20:00:19.846Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6407550Z     logger.go:146: 2024-03-28T20:00:20.579Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6407960Z     logger.go:146: 2024-03-28T20:00:21.524Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6408380Z     logger.go:146: 2024-03-28T20:00:22.598Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6408880Z     logger.go:146: 2024-03-28T20:00:23.154Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6409310Z     logger.go:146: 2024-03-28T20:00:23.927Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6409720Z     logger.go:146: 2024-03-28T20:00:24.550Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6410140Z     logger.go:146: 2024-03-28T20:00:25.862Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6410550Z     logger.go:146: 2024-03-28T20:00:27.188Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6410970Z     logger.go:146: 2024-03-28T20:00:28.418Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6411390Z     logger.go:146: 2024-03-28T20:00:29.459Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6411810Z     logger.go:146: 2024-03-28T20:00:30.213Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6412230Z     logger.go:146: 2024-03-28T20:00:31.365Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6412700Z     logger.go:146: 2024-03-28T20:00:32.104Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6413120Z     logger.go:146: 2024-03-28T20:00:33.057Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6413540Z     logger.go:146: 2024-03-28T20:00:33.755Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6413960Z     logger.go:146: 2024-03-28T20:00:34.906Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6414060Z     subscription_test.go:605: 
2024-03-28T20:02:13.6414530Z         	Error Trace:	/Users/runner/work/neo-go/neo-go/pkg/services/rpcsrv/subscription_test.go:605
2024-03-28T20:02:13.6414780Z         	Error:      	An error is expected but got nil.
2024-03-28T20:02:13.6415010Z         	Test:       	TestWSClientsLimit/default
2024-03-28T20:02:13.6415440Z     logger.go:146: 2024-03-28T20:00:35.802Z	INFO	shutting down RPC server	{"endpoint": "127.0.0.1:50911"}
2024-03-28T20:02:13.6415530Z === RUN   TestWSClientsLimit/8
2024-03-28T20:02:13.6416080Z     logger.go:146: 2024-03-28T20:00:36.071Z	INFO	initial gas supply is not set or wrong, setting default value	{"InitialGASSupply": "52000000"}
2024-03-28T20:02:13.6416760Z     logger.go:146: 2024-03-28T20:00:36.071Z	INFO	P2PNotaryRequestPayloadPool size is not set or wrong, setting default value	{"P2PNotaryRequestPayloadPoolSize": 1000}
2024-03-28T20:02:13.6417260Z     logger.go:146: 2024-03-28T20:00:36.071Z	INFO	MaxBlockSize is not set or wrong, setting default value	{"MaxBlockSize": 262144}
2024-03-28T20:02:13.6417830Z     logger.go:146: 2024-03-28T20:00:36.071Z	INFO	MaxBlockSystemFee is not set or wrong, setting default value	{"MaxBlockSystemFee": 900000000000}
2024-03-28T20:02:13.6418420Z     logger.go:146: 2024-03-28T20:00:36.071Z	INFO	MaxTransactionsPerBlock is not set or wrong, using default value	{"MaxTransactionsPerBlock": 512}
2024-03-28T20:02:13.6419040Z     logger.go:146: 2024-03-28T20:00:36.071Z	INFO	MaxValidUntilBlockIncrement is not set or wrong, using default value	{"MaxValidUntilBlockIncrement": 5760}
2024-03-28T20:02:13.6419410Z     logger.go:146: 2024-03-28T20:00:36.072Z	INFO	no storage version found! creating genesis block
2024-03-28T20:02:13.6419940Z     logger.go:146: 2024-03-28T20:00:36.075Z	INFO	ExtensiblePoolSize is not set or wrong, using default value	{"ExtensiblePoolSize": 20}
2024-03-28T20:02:13.6420450Z     logger.go:146: 2024-03-28T20:00:36.076Z	INFO	SessionPoolSize is not set or wrong, setting default value	{"SessionPoolSize": 20}
2024-03-28T20:02:13.6421030Z     logger.go:146: 2024-03-28T20:00:36.076Z	INFO	MaxIteratorResultItems is not set or wrong, setting default value	{"MaxIteratorResultItems": 100}
2024-03-28T20:02:13.6421570Z     logger.go:146: 2024-03-28T20:00:36.076Z	INFO	MaxFindResultItems is not set or wrong, setting default value	{"MaxFindResultItems": 100}
2024-03-28T20:02:13.6422070Z     logger.go:146: 2024-03-28T20:00:36.076Z	INFO	MaxNEP11Tokens is not set or wrong, setting default value	{"MaxNEP11Tokens": 100}
2024-03-28T20:02:13.6422630Z     logger.go:146: 2024-03-28T20:00:36.076Z	INFO	MaxRequestBodyBytes is not set or wong, setting default value	{"MaxRequestBodyBytes": 5242880}
2024-03-28T20:02:13.6423210Z     logger.go:146: 2024-03-28T20:00:36.076Z	INFO	MaxRequestHeaderBytes is not set or wong, setting default value	{"MaxRequestHeaderBytes": 1048576}
2024-03-28T20:02:13.6423570Z     logger.go:146: 2024-03-28T20:00:36.076Z	INFO	starting rpc-server	{"endpoint": "127.0.0.1:0"}
2024-03-28T20:02:13.6424000Z     logger.go:146: 2024-03-28T20:00:37.191Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6424560Z     logger.go:146: 2024-03-28T20:00:37.191Z	INFO	persisted to disk	{"blocks": 0, "keys": 127, "headerHeight": 0, "blockHeight": 0, "took": "111.875µs"}
2024-03-28T20:02:13.6424990Z     logger.go:146: 2024-03-28T20:00:38.295Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6425420Z     logger.go:146: 2024-03-28T20:00:39.385Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6425840Z     logger.go:146: 2024-03-28T20:00:40.522Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6426270Z     logger.go:146: 2024-03-28T20:00:41.525Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6426690Z     logger.go:146: 2024-03-28T20:00:42.501Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6427110Z     logger.go:146: 2024-03-28T20:00:43.491Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6427580Z     logger.go:146: 2024-03-28T20:00:44.385Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6428320Z     logger.go:146: 2024-03-28T20:00:44.823Z	ERROR	Error encountered with rpc request	{"code": -32603, "cause": "websocket users limit reached", "method": "", "params": "[]"}
2024-03-28T20:02:13.6428770Z     logger.go:146: 2024-03-28T20:00:45.188Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6429190Z     logger.go:146: 2024-03-28T20:00:45.603Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6429610Z     logger.go:146: 2024-03-28T20:00:45.959Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6430030Z     logger.go:146: 2024-03-28T20:00:46.343Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6430450Z     logger.go:146: 2024-03-28T20:00:46.834Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6430870Z     logger.go:146: 2024-03-28T20:00:47.265Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6431290Z     logger.go:146: 2024-03-28T20:00:47.647Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6431710Z     logger.go:146: 2024-03-28T20:00:48.261Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-03-28T20:02:13.6432100Z     logger.go:146: 2024-03-28T20:00:48.463Z	INFO	shutting down RPC server	{"endpoint": "127.0.0.1:50978"}
2024-03-28T20:02:13.6432190Z === RUN   TestWSClientsLimit/disabled
2024-03-28T20:02:13.6432750Z     logger.go:146: 2024-03-28T20:00:48.465Z	INFO	initial gas supply is not set or wrong, setting default value	{"InitialGASSupply": "52000000"}
2024-03-28T20:02:13.6433430Z     logger.go:146: 2024-03-28T20:00:48.465Z	INFO	P2PNotaryRequestPayloadPool size is not set or wrong, setting default value	{"P2PNotaryRequestPayloadPoolSize": 1000}
2024-03-28T20:02:13.6433930Z     logger.go:146: 2024-03-28T20:00:48.465Z	INFO	MaxBlockSize is not set or wrong, setting default value	{"MaxBlockSize": 262144}
2024-03-28T20:02:13.6434500Z     logger.go:146: 2024-03-28T20:00:48.465Z	INFO	MaxBlockSystemFee is not set or wrong, setting default value	{"MaxBlockSystemFee": 900000000000}
2024-03-28T20:02:13.6435080Z     logger.go:146: 2024-03-28T20:00:48.465Z	INFO	MaxTransactionsPerBlock is not set or wrong, using default value	{"MaxTransactionsPerBlock": 512}
2024-03-28T20:02:13.6435700Z     logger.go:146: 2024-03-28T20:00:48.465Z	INFO	MaxValidUntilBlockIncrement is not set or wrong, using default value	{"MaxValidUntilBlockIncrement": 5760}
2024-03-28T20:02:13.6436070Z     logger.go:146: 2024-03-28T20:00:48.467Z	INFO	no storage version found! creating genesis block
2024-03-28T20:02:13.6436600Z     logger.go:146: 2024-03-28T20:00:48.721Z	INFO	ExtensiblePoolSize is not set or wrong, using default value	{"ExtensiblePoolSize": 20}
2024-03-28T20:02:13.6437110Z     logger.go:146: 2024-03-28T20:00:48.721Z	INFO	SessionPoolSize is not set or wrong, setting default value	{"SessionPoolSize": 20}
2024-03-28T20:02:13.6437680Z     logger.go:146: 2024-03-28T20:00:48.721Z	INFO	MaxIteratorResultItems is not set or wrong, setting default value	{"MaxIteratorResultItems": 100}
2024-03-28T20:02:13.6438220Z     logger.go:146: 2024-03-28T20:00:48.721Z	INFO	MaxFindResultItems is not set or wrong, setting default value	{"MaxFindResultItems": 100}
2024-03-28T20:02:13.6438720Z     logger.go:146: 2024-03-28T20:00:48.721Z	INFO	MaxNEP11Tokens is not set or wrong, setting default value	{"MaxNEP11Tokens": 100}
2024-03-28T20:02:13.6439280Z     logger.go:146: 2024-03-28T20:00:48.721Z	INFO	MaxRequestBodyBytes is not set or wong, setting default value	{"MaxRequestBodyBytes": 5242880}
2024-03-28T20:02:13.6439850Z     logger.go:146: 2024-03-28T20:00:48.721Z	INFO	MaxRequestHeaderBytes is not set or wong, setting default value	{"MaxRequestHeaderBytes": 1048576}
2024-03-28T20:02:13.6440270Z     logger.go:146: 2024-03-28T20:00:48.721Z	INFO	starting rpc-server	{"endpoint": "127.0.0.1:0"}
2024-03-28T20:02:13.6441020Z     logger.go:146: 2024-03-28T20:00:49.113Z	ERROR	Error encountered with rpc request	{"code": -32603, "cause": "websocket users limit reached", "method": "", "params": "[]"}
2024-03-28T20:02:13.6441430Z     logger.go:146: 2024-03-28T20:00:49.455Z	INFO	shutting down RPC server	{"endpoint": "127.0.0.1:50989"}
2024-03-28T20:02:13.6442010Z     logger.go:146: 2024-03-28T20:00:49.455Z	INFO	persisted to disk	{"blocks": 0, "keys": 127, "headerHeight": 0, "blockHeight": 0, "took": "110.125µs"}
2024-03-28T20:02:13.6442120Z --- FAIL: TestWSClientsLimit (76.09s)
2024-03-28T20:02:13.6442290Z     --- FAIL: TestWSClientsLimit/default (62.70s)
2024-03-28T20:02:13.6442440Z     --- PASS: TestWSClientsLimit/8 (12.39s)
2024-03-28T20:02:13.6442610Z     --- PASS: TestWSClientsLimit/disabled (0.99s)
```

--- a/pkg/services/rpcsrv/subscription_test.go
+++ b/pkg/services/rpcsrv/subscription_test.go
@@ -602,7 +602,7 @@ func TestWSClientsLimit(t *testing.T) {
 				cfg.ApplicationConfiguration.RPC.MaxWebSocketClients = limit
 			})
 
-			dialer := websocket.Dialer{HandshakeTimeout: 5 * time.Second}
+			dialer := websocket.Dialer{HandshakeTimeout: 10 * time.Second}
 			url := "ws" + strings.TrimPrefix(httpSrv.URL, "http") + "/ws"
 			wss := make([]*websocket.Conn, effectiveClients)
 			var wg sync.WaitGroup

--- a/pkg/services/rpcsrv/subscription_test.go
+++ b/pkg/services/rpcsrv/subscription_test.go
@@ -4,6 +4,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 
@@ -604,25 +605,37 @@ func TestWSClientsLimit(t *testing.T) {
 			dialer := websocket.Dialer{HandshakeTimeout: 5 * time.Second}
 			url := "ws" + strings.TrimPrefix(httpSrv.URL, "http") + "/ws"
 			wss := make([]*websocket.Conn, effectiveClients)
+			var wg sync.WaitGroup
 
-			for i := 0; i < len(wss)+1; i++ {
-				ws, r, err := dialer.Dial(url, nil)
-				if r != nil && r.Body != nil {
-					defer r.Body.Close()
-				}
-				if i < effectiveClients {
+			// Dial effectiveClients connections in parallel
+			for i := 0; i < effectiveClients; i++ {
+				wg.Add(1)
+				j := i
+				go func() {
+					defer wg.Done()
+					ws, r, err := dialer.Dial(url, nil)
+					if r != nil {
+						defer r.Body.Close()
+					}
 					require.NoError(t, err)
-					wss[i] = ws
-					// Check that it's completely ready.
+					wss[j] = ws
 					doSomeWSRequest(t, ws)
-				} else {
-					require.Error(t, err)
-				}
+				}()
+			}
+
+			wg.Wait()
+
+			// Attempt one more connection, which should fail
+			_, r, err := dialer.Dial(url, nil)
+			require.Error(t, err, "The connection beyond the limit should fail")
+			if r != nil {
+				r.Body.Close()
 			}
 			// Check connections are still alive (it actually is necessary to add
 			// some use of wss to keep connections alive).
-			for i := 0; i < len(wss); i++ {
-				doSomeWSRequest(t, wss[i])
+			for _, ws := range wss {
+				doSomeWSRequest(t, ws)
+				ws.Close()
 			}
 		})
 	}

